experiments/prediction/environments/RandomWalk.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+from RlGlue import BaseEnvironment
```

Comment:
Do we need to use RlGlue?
If I want to use a code, I'm personally more comfortable if the code uses as few as possible things that I don't know what they are. Specifically for people outside of the UofA, I'm not sure if people know what RlGlue is supposed to do.

Reply:
I think we should use `RlGlue`. I have two reasons: (a) I want this codebase to be an example of how RL code should be written. Having hand-coded RL learning loops in every file is a really bad idea for reproducibility and will lead towards things like OpenAI Gym which uses a non-traditional learning loop by default. (b) Using a shared library also reduces the amount of code that we have to maintain and reduces the visual noise of the code. By moving some of this boilerplate into a hidden place, we have a higher proportion of code that is focused on our novel contribution / ideas.

Having a comment somewhere that cites Adam's MSc thesis and also having comments that quickly explain what `glue = RlGlue(agent, env)` as well as `glue.start()`, `glue.step()`, `glue.runEpisode(...)`, etc. will be useful though. I will add those.

Reply:
Great!